src/endpoints/auth/auth.controller.ts

```diff
@@ -202,3 +209,66 @@ export const codeEmailRouter: FastifyPluginAsync<FastifyPluginOptions> = async (
     }
   );
 };
+
+export const newPasswordEmailRouter: FastifyPluginAsync<FastifyPluginOptions> = async (server, options) => {
+  server.post<{ Body: IBodyNewPasswordEmail }>(
+    '/email/new-password',
+    {
+      schema: {
+        tags: [OpenAPITagsEnum.AUTH],
+        summary: 'New password email here',
+        body: {
+          type: 'object',
+          properties: {
+            email: { type: 'string', format: 'email', example: 'only@test.com' },
+            password: {
+              type: 'string',
+              minLength: 8,
+              maxLength: 256,
+              example: 'passwordTest',
+            },
+            code: { type: 'string', minLength: 6, maxLength: 6 },
+          },
+          required: ['email', 'password', 'code'],
+        },
+        response: {
+          200: newPasswordEmailOutputSchema,
+        },
+      },
+    },
+    async (req, reply) => {
+      const { email, code: receivedCode, password: newPassword } = req.body;
+
+      const user = await UserEntity.findOne({ email });
+
+      if (!user) {
+        throw createError(401, 'Email does not exist.');
+      }
+
+      const savedCode = await VerificationCodesEntity.findOne({ userId: String(user._id) });
+
+      VerificationCodeService.validateCode(savedCode, receivedCode);
```

Comment:
Nothing is happening here. You should either check the result or handle the possible error